### DISCUSSION TITLE
About Eww widgets

### DISCUSSION THREAD

**Discussion by Balssh** [prasanthrangan/hyprdots#480] [Ideas]
I've seen Eww is a wanted feature at some point. Due to its satanic config language, I think the best approach would be to get inspiration from something like
https://github.com/dharmx/vile#-showcase
https://github.com/linkfrg/dotfiles

What do you guys think?

**Comment by sitolam** [comment]
I like the second one

**Comment by kRHYME7** [comment]
How about Ags compared to eww?

**Reply by Balssh** [reply]
Right after posting this discussion I tried ags. Do you know how it’s better than eww, besides being easier to configure?
Also do you have some examples? What I tried is a bit buggy.

**Reply by kRHYME7** [reply]
I'm a bit not ready to commit to configuring those right now thu. I just saw that AGS got a bit of a spotlight on hyprland discord server atm. But IMO whatever are best fits and better integrates for this dotfiles should be better.  
I will provide screenshots very soon.

**Reply by Balssh** [reply]
Fair, I thought you saw/tried some specific dotfiles and you liked them. I tried Aylurs’ config, but it’s been a bit buggy.

**Reply by kRHYME7** [reply]
I guess Ags' is still in development. I can help on testing incase you implement something soon.

**Reply by Balssh** [reply]
I'll fool around with ags a bit. Though I'm not sure how to proceed, meaning what should I strive for? I was thinking maybe the bar, control center and maaaybe the powemenu? The launcher can remain rofi I guess.

**Reply by kRHYME7** [reply]
Me either, A Notification center is a  great start I guess or a tool bar something that has screenshot screen record volume etc.
I have swaync on my list now but maybe Eww or ags may be more customizable if I have the time to tinker with them.

**Reply by Balssh** [reply]
From the short time I had looked into Eww/Ags I gathere that:
- AGS should be easier to config given it's using JS instead of that ungodly lisp-based Yuck that Eww uses
- AGS may eat less resources than Eww
- if properly customized, the results are impressive, as you can fine-tune things more.
Ultimately I, given https://github.com/prasanthrangan/hyprdots/issues/486 I think I could focus on a bar that has following "widgets":
- workspaces/windows
- current media playing
- tray for network and bluetooth
- volume, brightness controls
- power button
- dropdown control center with memory/cpu/gpu/disk usages, current user, notifications list.

**Comment by T-Crypt** [comment]
@Balssh -- I've been trying to integrate EWW into these dotfiles for a while. Eww is painful as hell and so annoying to deal with (I hate yuck).

**Reply by Balssh** [reply]
Yeah, that’s why I’m more inclined towards AGS as at least there you can actually google js things.

**Comment by prasanthrangan** [comment]
The idea is to toggle between the `widget` mode and `classic` mode on the fly,  which will require porting each one of these configs to corresponding eww widget -

- rofi configs
  - app launcher
  - clipboard
  - themeselect
  - wall select
  - quickapps
- waybar
- logout menu
- dunst (if possible)
